eng/pipelines/templates/steps/build-and-test.yml

```diff
@@ -252,13 +252,31 @@ steps:
     - template: /eng/pipelines/templates/steps/retain-test-logs.yml
     - template: /eng/pipelines/templates/steps/retain-heap-dump-hprofs.yml
 
+  # The name of a test case in the results file isn't helpful in the web view.
+  # Update it to include the breadcrumb namespace and the type.
+  - pwsh: |
+      echo "Current directory: $(get-location)"
+      Get-ChildItem "TEST-*.xml" | ForEach-Object {
+      echo "Updating $($_.Name)"
+      [xml]$xml = Get-Content $_.FullName
+      $xml.SelectNodes("//testcase") | ForEach-Object {
+          $parts = $_.classname -split '\.'
+          $shortClassname = (($parts[0..($parts.Length -2)] | ForEach-Object { $_[0] }) -join ".") + ".$($parts[-1])"
+          $_.name = "$shortClassname.$($_.name)"
+          }
+          $xml.Save($_.FullName)
+      }
+    displayName: 'Update test case names for reporting'
+    workingDirectory: $(System.DefaultWorkingDirectory)/sdk
+
   - task: PublishTestResults@2
     condition: always()
     inputs:
       testResultsFormat: 'JUnit'
       testResultsFiles: '**/TEST-*.xml'
       searchFolder: '$(System.DefaultWorkingDirectory)/sdk'
       mergeTestResults: true
+      publishRunAttachments: true # just for debugging
```

Comment:
Not sure this actually did anything in DevOps

Reply:
it is true! It's just a mystery where they go so I thought I'd turn it on in case that wasn't true. :)